Basic script to produce CSV entity data

rboulton opened the pull request:
This script takes the output of the `script/export_abbreviation_entities` script in whitehall, de-dupes the entities in it, assigns IDs, and outputs the result in CSV format.

Commit:
Basic script to produce CSV entity data

This script takes the output of the
`script/export_abbreviation_entities` script in whitehall, de-dupes the
entities in it, assigns IDs, and outputs the result in CSV format.

dhwthompson commented on make_ids.py:
```diff
@@ -0,0 +1,35 @@
+        yield (unicode(i), json.dumps(entity["terms"]))
+
+
+def generate_entities(fobj):
```
I have no idea what a `fobj` is.

rboulton replied:
Added a docstring.

fobj is a common idiomatic name in python projects I've worked on for "file object" - I agree it needs explaining, though.

dhwthompson commented on make_ids.py:
```diff
@@ -0,0 +1,35 @@
+
+
+if __name__ == '__main__':
+    load_entities_from_file(sys.argv[1], sys.argv[2])
```
Since this script is using command-line arguments for filenames anyway, and so doesn’t need any filename logic, would it be reasonable for it to read from `stdin` and write to `stdout`, to make it more Unix-friendly?

rboulton replied:
It could do, but I don't want to refactor such things right now since I don't anticipate this script living in its current form for long and I'd rather work on functionality.  In particular, we'll be wanting to supply multiple input files in the near future, and also may not want to be constrained to reading input files in any particular order.

I also like being able to guarantee that the output file isn't clobbering an existing file.

dhwthompson replied:
Yup, that all sounds fair; thanks for clarifying.

dhwthompson commented:
I would find this much easier to understand if the functions had docstrings explaining what they did, what they expected to take in as parameters and what they returned as results.

Commit:
Add docstrings to all functions

rboulton commented:
I'd skipped documenting things because this is "just a quick hacky script that I don't expect to live for long". However, it's much better to start things as we mean to go on, and such scripts have a habit of living longer than we expect, so I've added full docstrings for all functions, and renamed a couple to be more clear.

Thanks for keeping me honest, @fatbusinessman.

dhwthompson commented:
> I've added full docstrings for all functions, and renamed a couple to be more clear.

Excellent: thank you very much. It’s much clearer to me what the script does now.
